app/api/routes.py

```diff
@@ -480,16 +517,21 @@ def create_resource(json, db):
         db.session.commit()
         index.save_object(new_resource.serialize_algolia_search)
 
-    except (AlgoliaUnreachableHostException, AlgoliaException) as e:
-        logger.exception(e)
-        print(f"Algolia failed to index new resource '{new_resource.name}'")
-
     except IntegrityError as e:
         logger.exception(e)
         return utils.standardize_response(status_code=422)
 
+    except (AlgoliaUnreachableHostException, AlgoliaException) as e:
+        if (environ.get("FLASK_ENV") != 'development'):
+            db.session.rollback()
+            logger.exception(e)
+            msg = f"Algolia failed to index new resource '{new_resource.name}'"
+            print(msg)
```

Comment:
Minor thing but I think you forgot a `print(msg)` there, also are we going to make it so that it saves resources that succeeds and return an error with the rest? or will it just fail the whole request and delete all the saved successful ones?

Reply:
You're right, that should be `logger.`something. I want it to log the name of the resource, might as well make it the same message that we send out to the client